freqtrade/strategy/strategyupdater.py

```diff
@@ -0,0 +1,253 @@
+    def visit_arguments(self, node):
+        if isinstance(node.args, list):
+            for arg in node.args:
+                arg.arg = self.check_dict(StrategyUpdater.name_mapping, arg.arg)
+        return node
+
+    def visit_Name(self, node):
+        # if the name is in the mapping, update it
+        node.id = self.check_dict(StrategyUpdater.name_mapping, node.id)
+        return node
+
+    def visit_Import(self, node):
+        # do not update the names in import statements
+        return node
+
+    def visit_ImportFrom(self, node):
+        # if hasattr(node, "module"):
+        #    if node.module == "freqtrade.strategy.hyper":
+        #        node.module = "freqtrade.strategy"
+        return node
+
+    def visit_If(self, node: ast_comments.If):
+        for child in ast_comments.iter_child_nodes(node):
+            self.visit(child)
+        return node
+
+    def visit_FunctionDef(self, node):
+        node.name = self.check_dict(StrategyUpdater.function_mapping, node.name)
+        self.generic_visit(node)
+        return node
+
+    def visit_Attribute(self, node):
+        if (
+                isinstance(node.value, ast_comments.Name)
+                and node.value.id == 'trades'
+                and node.attr == 'nr_of_successful_buys'
+        ):
+            node.attr = 'nr_of_successful_entries'
+        return node
+
+    def visit_ClassDef(self, node):
+        # check if the class is derived from IStrategy
+        if any(isinstance(base, ast_comments.Name) and
+               base.id == 'IStrategy' for base in node.bases):
+            # check if the INTERFACE_VERSION variable exists
+            has_interface_version = any(
+                isinstance(child, ast_comments.Assign) and
+                isinstance(child.targets[0], ast_comments.Name) and
+                child.targets[0].id == 'INTERFACE_VERSION'
+                for child in node.body
+            )
+
+            # if the INTERFACE_VERSION variable does not exist, add it as the first child
+            if not has_interface_version:
+                node.body.insert(0, ast_comments.parse('INTERFACE_VERSION = 3').body[0])
+            # otherwise, update its value to 3
+            else:
+                for child in node.body:
+                    if (
+                            isinstance(child, ast_comments.Assign)
+                            and isinstance(child.targets[0], ast_comments.Name)
+                            and child.targets[0].id == 'INTERFACE_VERSION'
+                    ):
+                        child.value = ast_comments.parse('3').body[0].value
+        self.generic_visit(node)
+        return node
+
+    def visit_Subscript(self, node):
+        if isinstance(node.slice, ast_comments.Constant):
+            if node.slice.value in StrategyUpdater.rename_dict:
+                # Replace the slice attributes with the values from rename_dict
+                node.slice.value = StrategyUpdater.rename_dict[node.slice.value]
```

Comment:
What scenario would this cover?

I've noticed that tests don't hit it - which stuns me, as i think there's tests for (almost) everything now.